share/ext/spirv-tools/build_spirv-tools.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+# License: MIT
+# Copyright 2022 The Defold Foundation
+
+PLATFORM=$1
+PWD=$(pwd)
+SOURCE_DIR=${PWD}/source
+BUILD_DIR=${PWD}/build/${PLATFORM}
+
+if [ -z "$PLATFORM" ]; then
+    echo "No platform specified!"
+    exit 1
+fi
+
+CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=Release ${CMAKE_FLAGS}"
+
+case $PLATFORM in
+    arm64-macos)
+        CMAKE_FLAGS="-DCMAKE_OSX_ARCHITECTURES=arm64 ${CMAKE_FLAGS}"
+        ;;
+    x86_64-macos)
+        CMAKE_FLAGS="-DCMAKE_OSX_ARCHITECTURES=x86_64 ${CMAKE_FLAGS}"
+        ;;
+esac
+
+if [ ! -d "${SOURCE_DIR}" ]; then
+    git clone https://github.com/KhronosGroup/SPIRV-Tools.git ${SOURCE_DIR}
+fi
+
+python ${SOURCE_DIR}/utils/git-sync-deps
+
+# Build
+
+mkdir -p ${BUILD_DIR}
+
+pushd $BUILD_DIR
+
+echo "CMAKE_FLAGS: '${CMAKE_FLAGS}"
+cmake ${CMAKE_FLAGS} ${SOURCE_DIR}
+cmake --build . --config Release
+
+EXE_SUFFIX=
+case $PLATFORM in
+    win32|x86_64-win32)
+        EXE_SUFFIX=.exe
+        SRC_EXE=./tools/Release/spirv-opt${EXE_SUFFIX}
+        ;;
+    *)
+        SRC_EXE=./tools/spirv-opt${EXE_SUFFIX}
+        ;;
+esac
```

Comment:
For future reference maybe we should add a comment that this *only* packages the spirv-opt tool. I think there might be compile options for which items to build, but I didn't spend that much time optimizing this setup..